src/Compilers/Test/Core/Compilation/ControlFlowGraphVerifier.cs

```diff
@@ -1045,6 +1097,12 @@ bool isLongLivedCaptureReferenceSyntax(SyntaxNode captureReferenceSyntax)
                             }
 
                             CSharpSyntaxNode parent = syntax.Parent;
+                            while (parent is PostfixUnaryExpressionSyntax { OperatorToken: { RawKind: (int)CSharp.SyntaxKind.ExclamationToken } })
```

Comment:
>while (parent is PostfixUnaryExpressionSyntax { OperatorToken: { RawKind: (int)CSharp.SyntaxKind.ExclamationToken } }) [](start = 28, length = 118)

I think this logic should be incorporated into ```applyParenthesizedIfAnyCS``` and the helper renamed accordingly